src/main/scala/mesosphere/marathon/state/RunSpec.scala

```diff
@@ -64,4 +65,32 @@ trait RunSpec extends plugin.RunSpec {
   val unreachableStrategy: UnreachableStrategy
   val killSelection: KillSelection
   val networks: Seq[Network]
+
+  final def ref: RunSpecRef = RunSpecRef(kind, id, version)
```

Comment:
We don't absolutely need it now but it goes along with configRef. There's lots of places where we reference a runSpecId + version. I implemented configRef for the rateLimiter (it was clearer than using a tuple IMO).

Having a configRef instead of a RunSpecRef is useful because it prevents accidentally using the wrong version to refer to runSpec. In the case of rateLimiter this was helpful as it made it more clear that it issues rate limits for a configRef, and not for a runSpecRef.

I don't think we allow pods and apps to have the same id so we can probably get rid of "kind", on second glance.

Reply:
I'm going to remove the kind field :)